include/nbl/builtin/hlsl/spirv_intrinsics/glsl.std.450.hlsl

```diff
@@ -94,63 +112,173 @@ float32_t4 unpackSnorm4x8(uint32_t p);
-enable_if_t<is_floating_point_v<FloatingPointVector> && is_vector_v<FloatingPointVector>, FloatingPointVector> normalize(FloatingPointVector vec);
+T normalize(T vec);
 
-// TODO: will not work for vectors, fix
-template<typename FloatingPoint>
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450FClamp, "GLSL.std.450")]]
-enable_if_t<is_floating_point_v<FloatingPoint>, FloatingPoint> fClamp(FloatingPoint val, FloatingPoint min, FloatingPoint max);
-template<typename UnsignedInteger>
+T fClamp(T val, T _min, T _max);
+template<typename T NBL_FUNC_REQUIRES(concepts::UnsignedIntegralVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450UClamp, "GLSL.std.450")]]
-enable_if_t<is_integral_v<UnsignedInteger> && !is_signed_v<UnsignedInteger>, UnsignedInteger> uClamp(UnsignedInteger val, UnsignedInteger min, UnsignedInteger max);
-template<typename Integer>
+T uClamp(T val, T _min, T _max);
+template<typename T NBL_FUNC_REQUIRES(concepts::IntegralVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450SClamp, "GLSL.std.450")]]
-enable_if_t<is_integral_v<Integer> && is_signed_v<Integer>, Integer> sClamp(Integer val, Integer min, Integer max);
+T sClamp(T val, T _min, T _max);
 
-template<typename FloatingPoint>
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450FMin, "GLSL.std.450")]]
-enable_if_t<is_floating_point_v<FloatingPoint>, FloatingPoint> fMin(FloatingPoint val);
-template<typename UnsignedInteger>
+T fMin(T val);
+template<typename T NBL_FUNC_REQUIRES(concepts::UnsignedIntegralVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450UMin, "GLSL.std.450")]]
-enable_if_t<is_integral_v<UnsignedInteger> && !is_signed_v<UnsignedInteger>, UnsignedInteger> uMin(UnsignedInteger val);
-template<typename Integer>
+T uMin(T val);
+template<typename T NBL_FUNC_REQUIRES(concepts::IntegralVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450SMin, "GLSL.std.450")]]
-enable_if_t<is_integral_v<Integer>&& is_signed_v<Integer>, Integer> sMin(Integer val);
+T sMin(T val);
 
-template<typename FloatingPoint>
+template<typename T NBL_FUNC_REQUIRES(concepts::FloatingPointVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450FMax, "GLSL.std.450")]]
-enable_if_t<is_floating_point_v<FloatingPoint>, FloatingPoint> fMax(FloatingPoint val);
-template<typename UnsignedInteger>
+T fMax(T val);
+template<typename T NBL_FUNC_REQUIRES(concepts::UnsignedIntegralVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450UMax, "GLSL.std.450")]]
-enable_if_t<is_integral_v<UnsignedInteger> && !is_signed_v<UnsignedInteger>, UnsignedInteger> uMax(UnsignedInteger val);
-template<typename Integer>
+T uMax(T val);
+template<typename T NBL_FUNC_REQUIRES(concepts::IntegralVectorOrScalar<T>)
 [[vk::ext_instruction(GLSLstd450SMax, "GLSL.std.450")]]
-enable_if_t<is_integral_v<Integer>&& is_signed_v<Integer>, Integer> sMax(Integer val);
+T sMax(T val);
```

Comment:
I think I know why your min and max fails, they take two args and your intrinsic only declares 1 XD